packages/core-data/src/reducer.js

```diff
@@ -521,6 +521,11 @@ export function userPermissions( state = {}, action ) {
 				...state,
 				[ action.key ]: action.isAllowed,
 			};
+		case 'RECEIVE_USER_PERMISSIONS':
+			return {
+				...state,
+				...action.permissions,
```

Comment:
Should we keep the existing permissions? I'd argue that we shouldn't, because if a user loses a permission in the meantime, it should be removed from the list of permissions.

```suggestion
				...action.permissions,
```

Reply:
I'm not sure I'm following here. Why would be dump the existing permissions from other resources? Also let's keep the logic the same as the singular form of the action in this PR, I don't want there to be any changes other than performance.

Reply:
Clearing the entire permissions state would be a mistake, because there are permissions for other resources.

The point is probably that if an `OPTIONS` request (or equivalent one) returns a header:
```
Allow: GET
```
then it means not only that there is a `read: true` permission, but that there is also `update: false`, `create: false`, `delete: false`. But we already take care of that, in the `getUserPermissionsFromAllowHeader` function.

Reply:
Ah, I see. Somehow I thought this retrieves them all at ince, which isn't the case, obviously.

Reply:
It retrieves all permissions at once for a query, but there are other queries or direct `canUser` calls, which all are stored in the same state, keyed by `actions` and `resourceEntity`.